bliss/models/galaxy_net.py

```diff
@@ -197,26 +288,210 @@ def plot_reconstruction(
-        images, background = batch["images"], batch["background"]
-        recon_mean = self(images, background)
+    def _plot_grid_examples(self, outputs):
+        images, recon_mean = outputs["images"], outputs["recon_mean"]
+        # 1.  plot a grid of all input images and recon_mean
+        # 2.  only plot the highest band
+
+        nrow = 16
         residuals = (images - recon_mean) / torch.sqrt(images)
-        self.log("max_residual", residuals.abs().max())
+
+        image_grid = make_grid(images, nrow=nrow)[0]
+        recon_grid = make_grid(recon_mean, nrow=nrow)[0]
+        residual_grid = make_grid(residuals, nrow=nrow)[0]
+        h, w = image_grid.size()
+        base_size = 8
+        fig = plt.figure(figsize=(3 * base_size, int(h / w * base_size)))
+        for i, grid in enumerate([image_grid, recon_grid, residual_grid]):
+            plt.subplot(1, 3, i + 1)
+            plt.imshow(grid.cpu().numpy(), interpolation=None)
+            if i == 0:
+                plt.title("images")
+            elif i == 1:
+                plt.title("recon_mean")
+            else:
+                plt.title("residuals")
+            plt.xticks([])
+            plt.yticks([])
+        return fig
+
+
+class OneCenteredGalaxyEncoder(nn.Module):
+    def __init__(self, main_encoder, main_decoder, residual_encoder, slen=None, allow_pad=False):
+        super().__init__()
+        self.main_encoder = main_encoder
+        self.main_decoder = main_decoder
+        self.residual_encoder = residual_encoder
+        self.slen = slen
+        self.allow_pad = allow_pad
+
+    def forward(self, image, background=0):
+        assert image.shape[-2] == image.shape[-1]
+        if self.allow_pad:
+            if image.shape[-1] < self.slen:
+                d = self.slen - image.shape[-1]
+                lpad = d // 2
+                upad = d - lpad
+                min_val = image.min()
+                image = F.pad(image, (lpad, upad, lpad, upad), value=min_val)
+                if isinstance(background, torch.Tensor):
+                    background = F.pad(background, (lpad, upad, lpad, upad))
+        latent_main = self.main_encoder(image - background)
+        recon_mean_main = F.relu(self.main_decoder(latent_main)) + background
+        latent_residual = self.residual_encoder(image - recon_mean_main)
+        return torch.cat((latent_main, latent_residual), dim=-1)
+
+
+class OneCenteredGalaxyDecoder(nn.Module):
+    def __init__(self, main_decoder, residual_decoder):
+        super().__init__()
+        self.main_decoder = main_decoder
+        self.residual_decoder = residual_decoder
+
+    def forward(self, latent):
+        latent_main, latent_residual = torch.split(latent, latent.shape[-1] // 2, -1)
+        recon_mean_main = F.relu(self.main_decoder(latent_main))
+        recon_mean_residual = self.residual_decoder(latent_residual)
+        return F.relu(recon_mean_main + recon_mean_residual)
+
+
+class CenteredGalaxyEncoder(nn.Module):
```

Comment:
It might be cool to have some documentation for `CenteredGalaxyEncoder`, `CenteredGalaxyDecoder` and `ResidualConvBlock` just explaining what they do at a high level and descriptions of some of the more unintuitive arguments. A couple of comments here and there for some of the more complicated constructions might be helpful. 

I don't think this should stop you from merging, it would just help me understand better how you construct this neural network and some of the modelling choices you made, which would also be helpful for writing the paper :)